src/movies.js

```diff
@@ -1,15 +1,105 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
-
+function getAllDirectors(array) {
+    let directorsArr = array.map((movie) => {
+        return movie.director
+    })
+    return directorsArr
+}
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies (array){
+    let dramaMovies = array.filter ((movie) => {
+        return movie.director ==="Steven Spielberg" && movie.genre.includes("Drama")
+    })
+        return dramaMovies.length
+}
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage (movies){
+    if(movies.length===0){
+        return 0
+
+    }else{
+    let total = movies.reduce ((acc, movies) =>{
+        if ( movies.rate == undefined){
+            return acc
+        } else {
+        return acc + movies.rate }
+    },0)
+    let average = total/ movies.length
+    average= average.toFixed(2)
+    average = Number(average)
+    return average
+}
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate (movies) {
+    if (!movies.includes("Drama")){
```

Comment:
since movies is an array of objects, then this condition will try to compare the string "Drama" to each object, and so, the condition will always come out as true (!false). So the function will return 0 for every case. It seems you are already checking the "Drama" part later so no need for this par of the code.